src/Headers.php

```diff
@@ -58,58 +58,49 @@ class Headers implements Countable, Iterator
      * will be lazy loaded)
      *
      * @param  string $string
-     * @param  string $EOL EOL string; defaults to {@link EOL}
+     * @param  string $EOL DEPRECATED
      * @throws Exception\RuntimeException
      * @return Headers
      */
-    public static function fromString($string, $EOL = self::EOL)
+    public static function fromString($string, $EOL = null)
     {
         $headers     = new static();
         $currentLine = '';
         $emptyLine   = 0;
 
         // iterate the header lines, some might be continuations
-        $lines = explode($EOL, $string);
-        $total = count($lines);
-        for ($i = 0; $i < $total; $i += 1) {
-            $line = $lines[$i];
-
+        foreach (preg_split('/\R/', $string) as $line) {
```

Comment:
WHOA!!! NO!!!!

This does *NOT* follow RFC 2822, and introduces potential for security vectors. Please read [RFC 2822, section 2.1](https://tools.ietf.org/html/rfc2822#section-2.1), which states:

> Messages are divided into lines of characters.  A line is a series of characters that is delimited with the two characters carriage-return and line-feed; that is, the carriage return (CR) character (ASCII value 13) followed immediately by the line feed (LF) character (ASCII value 10).  (The carriage-return/line-feed pair is usually written in this document as "CRLF".)

Additionally, you are undoing a *reported* vulnerability ([ZF2015-04](http://framework.zend.com/security/advisory/ZF2015-04)).

`\R` matches *any* of `\r`, `\n`, or `\r\n`; we specifically were matching only `\r\n` previously to be compliant with the specification and prevent message splitting.